src/transport_interface/transport_interface_test.c

```diff
@@ -421,26 +522,137 @@ static void prvSendRecvCompareFunc( void * pParam )
     for( testBufferSize = 1; testBufferSize <= TRANSPORT_TEST_BUFFER_WRITABLE_LENGTH; testBufferSize = testBufferSize + 1 )
     {
         /* Initialize the test data buffer. */
-        prvInitializeTestData( pTrasnportTestBufferStart, testBufferSize );
+        prvInitializeTestData( pTransportTestBufferStart, testBufferSize );
 
         /* Send the test data to the server. */
         if( pThreadParameter->xResult == true )
         {
-            pThreadParameter->xResult = prvTransportSendData( pTestTransport, pNetworkContext, pTrasnportTestBufferStart,
+            pThreadParameter->xResult = prvTransportSendData( pTestTransport,
+                                                              pNetworkContext,
+                                                              pTransportTestBufferStart,
+                                                              testBufferSize );
+        }
+
+        /* Receive the test data from server. */
+        if( pThreadParameter->xResult == true )
+        {
+            pThreadParameter->xResult = prvTransportRecvData( pTestTransport,

```

Comment:
what happens if we don't receive all of the data back in one call? like what if the response is in more than one packet? I'm worried that on one iteration of the loop we receive only a part of the response then on the next iteration we receive the rest of it when we expect new data entirely.

Reply:
If we do not receive any data (or receive partial data) the test fails.
The next test is not affected as the test setup creates a new connection.